builtin/credential/aws/path_login.go

```diff
@@ -384,6 +384,11 @@ func (b *backend) verifyInstanceMeetsRoleRequirements(ctx context.Context,
 		return nil, fmt.Errorf("nil identityDoc")
 	}
 
+	// Verify that the instance ID matches one of the ones set by the role
+	if len(roleEntry.BoundEc2InstanceIDs) > 0 && !strutil.StrListContains(roleEntry.BoundEc2InstanceIDs, *instance.InstanceId) {
```

Comment:
Any reason you explicitly dereference the pointer here and in the next line?

Reply:
I dereference it here so I don't get a compiler error:

```bash
$ make test TEST=./builtin/credential/aws/
go generate 
# github.com/hashicorp/vault/builtin/credential/aws
/.../go/src/github.com/hashicorp/vault/builtin/credential/aws/path_login.go:388:111: cannot use instance.InstanceId (type *string) as type string in argument to strutil.StrListContains
FAIL    github.com/hashicorp/vault/builtin/credential/aws [build failed]
Makefile:34: recipe for target 'test' failed
make: *** [test] Error 2
```
I dereference it in the next line so that it shows up as expected in the error message.

Reply:
Oh, forgot about how it's all pointers in AWS land. Sounds good!